probe/sniff/sniffer_test.go

```diff
@@ -77,7 +72,16 @@ func TestMerge(t *testing.T) {
 			},
 		},
 		NodeMetadatas: report.NodeMetadatas{
-			srcEndpointNodeID: report.MakeNodeMetadata(),
+			srcEndpointNodeID: {
+				Metadata:  map[string]string{},
+				Counters:  map[string]int{},
+				Adjacency: report.MakeIDList(dstEndpointNodeID),
+			},
+			dstEndpointNodeID: report.NodeMetadata{
+				Metadata:  map[string]string{},
+				Counters:  map[string]int{},
+				Adjacency: report.MakeIDList(),
+			},
```

Comment:
Please add constructors for this initialization style. If we leave it as-is, and later add a field to NodeMetadata that needs initialization, it's very easy to forget to declare it in this callsite, and get very subtle runtime breakage.

I'm not at all happy with this pattern, but I'm not aware of a better way to do it. Suggestions welcome.